### PR TITLE
10-bit addressing not supported by I2C slave driver for STM32 target #51060

### PR DESCRIPTION
**Pull request by AshwiniMShinde-eaton**
I2C: Add 10-bit addressing support for I2C slave driver for STM32 target.
	This patch adds changes that will help to register STM32 target as a slave with 10-bit addressing enabled

### CODE REVIEW

File: drivers/i2c/i2c_ll_stm32_v2.c
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32_v2.c:227]: Since https://github.com/zephyrproject-rtos/zephyr/pull/43166, you should now use I2C_TARGET_FLAGS_ADDR_10_BITS

File: drivers/i2c/i2c_ll_stm32_v2.c
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32_v2.c:236]: Please fix indentation:
```suggestion
		if (data->slave_cfg->flags == I2C_SLAVE_FLAGS_ADDR_10_BITS)	{
			LL_I2C_SetOwnAddress1(i2c, config->address, LL_I2C_OWNADDRESS1_10BIT);
			LOG_DBG("i2c: slave #1 registered with 10-bit address");
		} else {
			LL_I2C_SetOwnAddress1(i2c, config->address << 1U, LL_I2C_OWNADDRESS1_7BIT);
			LOG_DBG("i2c: slave #1 registered with 7-bit address");
		}
```

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for sharing. Some parts to fix.

Also, it would be nice to return an error in case 10bits address mode is tried to be configured for slave2 as it is not supported. For instance, in slave 2 code, add:
```
		if (data->slave_cfg->flags == I2C_SLAVE_FLAGS_ADDR_10_BITS)	{
			return -EVINVAL;
		}
```

**Comment by erwango** [comment]
Your pull request does not conform with Zephyr contribution guidelines. For reference, I'd advise you to check:
- [Contribution guidelines on docs.zephyrproject.org](https://docs.zephyrproject.org/latest/contribute/guidelines.html)
- [Great presentation dealing with "the best way to get your patches merged"](https://youtu.be/mnVNLM7ynMQ) and the related [Slides](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf)
- Coding Guidelines bot results reported below.

Please:
- Check your commit message
- Remove the merge commit

File: drivers/i2c/i2c_ll_stm32_v2.c
Comment (AshwiniMShinde-eaton) [reply drivers/i2c/i2c_ll_stm32_v2.c]: Fixed

**Comment by AshwiniMShinde-eaton** [comment]
> Your pull request does not conform with Zephyr contribution guidelines. For reference, I'd advise you to check:
> 
> * [Contribution guidelines on docs.zephyrproject.org](https://docs.zephyrproject.org/latest/contribute/guidelines.html)
> * [Great presentation dealing with "the best way to get your patches merged"](https://youtu.be/mnVNLM7ynMQ) and the related [Slides](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf)
> * Coding Guidelines bot results reported below.
> 
> Please:
> 
> * Check your commit message. It should conform to the following template:
> 
> ```
> <dir>: <dir>: <Title summarizing the change>
> 
> <A description useful to the reader>
> 
> Signed-off-by: foo <foo@bar.baz>
> ```
> 
> * Remove the merge commit(s)
> * We don't merge fixup commits. Please squash all fixes in the corresponding commits

I've removed my previous commits and pushed the commit that follows all the guidelines. The checks are waiting for approval. Can you please approve?

**Review by erwango — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

File: drivers/i2c/i2c_ll_stm32_v2.c
Comment (teburd) [inline drivers/i2c/i2c_ll_stm32_v2.c:229]: The verbage for i2c has changed from master/slave to controller/target, while I don't expect the entire driver to be reworked for this a log message should probably use the newer friendlier terminology.

**Review by teburd — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_ll_stm32_v1.c
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32_v1.c:508]:
```suggestion
		return -ENOTSUP;
```

File: drivers/i2c/i2c_ll_stm32_v1.c
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32_v1.c:511]:
```suggestion
	}
	LL_I2C_SetOwnAddress1(i2c, config->address << 1U, LL_I2C_OWNADDRESS1_7BIT);
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by teburd — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by erwango** [comment]
@AshwiniMShinde-eaton Please fix compliance issue reported

**Comment by AshwiniMShinde-eaton** [comment]
> @AshwiniMShinde-eaton Please fix compliance issue reported

I've fixed it.

**Comment by AshwiniMShinde-eaton** [comment]
@erwango @teburd @FRASTM  Is this PR good to merge once all the checks pass?
I just wanted confirmation from you all that there are no further changes required from my side

**Review by teburd — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Comment by AshwiniMShinde-eaton** [comment]
> Please don't force push unless you are making changes, it removes the CI check passes and approvals

I've been rebasing(Using rebase branch button here) from time to time as it shows "This branch is out-of-date" with the base branch. When I get all the approvals and all checks pass then merging gets blocked as it requires rebasing. This cycle goes on and its been long time I haven't been able to get my PR merged. Is this the right way? Is there any way to avoid this or did I miss anything?

**Review by erwango — APPROVED** [review]

**Comment by prayassamriya** [comment]
> 

@AshwiniMShinde-eaton, May be you can request primary approvers/module owners to merge if they think it meets their expectations. due to time zone difference this issue may be surfacing again and again. what's your thoughts @FRASTM , @erwango and others?

**Comment by teburd** [comment]
> > Please don't force push unless you are making changes, it removes the CI check passes and approvals
> 
> I've been rebasing(Using rebase branch button here) from time to time as it shows "This branch is out-of-date" with the base branch. When I get all the approvals and all checks pass then merging gets blocked as it requires rebasing. This cycle goes on and its been long time I haven't been able to get my PR merged. Is this the right way? Is there any way to avoid this or did I miss anything?

So long as there are no conflicts, its ok.

**Review by teburd — APPROVED** [review]

**Comment by AshwiniMShinde-eaton** [comment]
> > > Please don't force push unless you are making changes, it removes the CI check passes and approvals
> > 
> > 
> > I've been rebasing(Using rebase branch button here) from time to time as it shows "This branch is out-of-date" with the base branch. When I get all the approvals and all checks pass then merging gets blocked as it requires rebasing. This cycle goes on and its been long time I haven't been able to get my PR merged. Is this the right way? Is there any way to avoid this or did I miss anything?
> 
> So long as there are no conflicts, its ok.

@teburd Could you(or anyone who is authorized to merge this PR) please help to get this merged?

**Comment by teburd** [comment]
> > > > Please don't force push unless you are making changes, it removes the CI check passes and approvals
> > > 
> > > 
> > > I've been rebasing(Using rebase branch button here) from time to time as it shows "This branch is out-of-date" with the base branch. When I get all the approvals and all checks pass then merging gets blocked as it requires rebasing. This cycle goes on and its been long time I haven't been able to get my PR merged. Is this the right way? Is there any way to avoid this or did I miss anything?
> > 
> > 
> > So long as there are no conflicts, its ok.
> 
> @teburd Could you(or anyone who is authorized to merge this PR) please help to get this merged?

When the release managers see this is ready to merge, which means there are at least 2 approvals including the assignee, and green CI, they merge it. There's filters the release managers use to periodically find PRs ready to merge and do so.

**Pull request merged**
